main.js

```diff
@@ -13,6 +13,275 @@ toggle.addEventListener('click', () => {
+    heading: 'My Recent Works',
+  },
+  {
+    class: ['img-hdr', 'langs'],
+    divCls: [
+      'card cd1',
+      'card cd2',
+      'card cd3',
+      'card cd4',
+      'card cd5',
+      'card cd6',
+    ],
+    listcls: 'cd-links',
+    heading: ['Multi-Post Stories', 'Gain+Glory'],
+    lists: ['Ruby on Rails', 'css', 'Javascript', 'html'],
+    btn: 'See Project',
+    btnCls: 'cardBtn',
+  },
+];
+const popupData = [
+  {
+    id: 'popup',
+    divCls: ['popimg', 'poptext', 'popHead', 'topBtns', 'btns'],
+    classes: [
+      'fa-solid fa-x close',
+      'imgMob',
+      'imgDesk',
+      'popTtl',
+      'btn seeLive',
+      'btn seeLink',
+      'popLinks',
+      'link',
+      'popPara',
+    ],
+    btns: ['See Live', 'See source'],
+    images: ['./assets/mobile.png', './assets/popImg.png'],
+    h2: 'Keeping track of hundreds of components',
+    btnLinks: [
+      'https://prg-04.github.io/Portfolio_',
+      'https://github.com/prg-04/Portfolio_',
+    ],
+    lists: [
+      'Codekit',
+      'Github',
+      'Javascript',
+      'Bootstrap',
+      'Terminal',
+      'Codepen',
+    ],
+    para: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the releaLorem Ipsum is simply dummy text of the printing and typesetting industry, Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it 1960s with the release",
+  },
+];
+
+function createImg(src) {
+  const img = document.createElement('img');
+  img.src = src;
+  return img;
+}
+
+function createDiv(cls) {
+  const div = document.createElement('div');
+  div.className = cls;
+  return div;
+}
+
+function createDiv2(id) {
+  const div = document.createElement('div');
+  div.id = id;
+  return div;
+}
+
+function createH3(txt) {
+  const h3 = document.createElement('h3');
+  h3.innerText = txt;
+  return h3;
+}
+function createH2(txt) {
+  const h2 = document.createElement('h2');
+  h2.innerText = txt;
+  return h2;
+}
+
+function createUl(cls) {
+  const ul = document.createElement('ul');
+  ul.className = cls;
+  return ul;
+}
+
+function createLi(txt) {
+  const li = document.createElement('li');
+  li.innerText = txt;
+  return li;
+}
+
+function createLists(arr) {
+  const lists = arr.map((elem) => {
+    const li = createLi(elem);
+    return li;
+  });
+  return lists;
+}
+
+function createPara(txt) {
+  const p = document.createElement('p');
+  p.innerText = txt;
+  return p;
+}
+
+function createBtn(txt) {
+  const btn = document.createElement('button');
+  btn.innerText = txt;
+  return btn;
+}
+
+const work = document.getElementById('portfolio');
+
+const title = createDiv(cardsData[0].class);
+const h2 = createH2(cardsData[0].heading);
+title.appendChild(h2);
+work.appendChild(title);
```

Comment:
- [ ] The names you give your variables matter a lot as they can help your coding partner quickly understand what your code is doing. The variable `title` could have been qualified as what kind of title, e.g `const workSectionTitle = .....`

Across your codebase, I see a lot of code that cannot easily make me understand what you are trying to achieve. I believe you can refactor your code and keep it simple. As a professional software engineer, you will work with many teams in your career, and writing easy-to-understand code will make you an effective team member. You may need to go over the [javascript best practices](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).